scala-thoughts.md

```diff
@@ -0,0 +1,75 @@
+# Reflections on Learning Scala
+
+Scala has been an interesting language to start to learn. What stood out to me
+the most when I was learning it was that the syntax was super flexible - for
+me that was actually very challenging to handle at first. I think in the end I
+tried to just avoid the cases that seemed ambiguous to me, choosing to write
+
+```
+list1.zip(list2).drop(3)
+```
+
+in place of
+
+```
+list1 zip list2 drop 3
+```
+
+I don't think this flexibility is a bad thing (more thoughts on this later),
+but I would note that it has been a long time since I learned a language with
+even a slightly different syntax structure. Its surprising how much a few small
+decisions can do to complicate that structure.
+
+Regarding the functional parts of Scala, I found using them to be very
+satisfying, mostly because I've done enough C++11 and Java to want that
+expressiveness and written enough Python/Haskell/Rust to be comfortable
+using it.
+
+I could imagine that transitioning to Scala from an only-Java perspective would
+be much more challenging.
+
```

Comment:
So it sounds like the imperative-style syntax is more natural, but some background in functional languages helps with understanding how things work under the hood. Good! I agree that Scala's flexibility helps with people who have preferences for either programming style, but without a functional background it would be challenging, yep.